src/transport/pci/bus.rs

```diff
@@ -238,6 +242,10 @@ impl<C: ConfigurationAccess> PciRoot<C> {
                 .read_word(device_function, BAR0_OFFSET + 4 * bar_index),
         );
 
+        if size_mask == 0 {
+            return Ok(None);
```

Comment:
We still need to restore the original BAR value and command register configuration, which returning early skips. Maybe the "Restore the original value" step could be moved up before this early return? But the command register IO/mem decode disabling + re-enabling still needs to be wrapped around all BAR modifications, including the writing of the high half of 64-bit BARs below, so it's a bit trickier to rearrange (unless you just want to duplicate the `self.set_command()` call in the early return branch).

It would probably be good to verify that everything is put back the way it was originally in the unit tests as well, so this kind of thing would be caught.

Reply:
(I guess restoring the original BAR value doesn't really matter, since it should be all 0 bits in this case, but the point about the command register still stands.)